src/core/version_neg.c

```diff
@@ -394,135 +265,113 @@ QuicVersionNegotiationExtEncodeVersionNegotiationInfo(
             DesiredVersionsListLength = ARRAYSIZE(DefaultSupportedVersionsList);
         }
         //
-        // Generate Server VNI.
+        // Generate Server Version Info.
         //
-        VNILen = sizeof(uint32_t) + QuicVarIntSize(DesiredVersionsListLength) +
-            (DesiredVersionsListLength * sizeof(uint32_t));
+        VILen = sizeof(uint32_t) + (DesiredVersionsListLength * sizeof(uint32_t));
+        CXPLAT_DBG_ASSERT((DesiredVersionsListLength * sizeof(uint32_t)) + sizeof(uint32_t) > DesiredVersionsListLength + sizeof(uint32_t));
```

Comment:
The code all generally looks fine, but we need to make sure we guarantee things like this through out all this code. How do you know this can't overflow? Please do a quick pass at all your calculations like these.